game-core/src/main/java/games/strategy/triplea/settings/ClientSettingSwingUiBinding.java

```diff
@@ -234,7 +236,17 @@
       "Specifies host and port for connecting to a test HTTP lobby.") {
     @Override
     public SelectionComponent<JComponent> newSelectionComponent() {
-      return textField(ClientSetting.httpLobbyUriOverride);
+      return textField(
+          ClientSetting.httpLobbyUriOverride,
+          Object::toString,
+          encodedValue -> {
```

Comment:
While it looks like this code is identical to that in `UriClientSetting#decodeValue()`, it was a conscious decision to not try to reuse it.  The reasoning is this code defines a `String`-to-`URI` conversion in the UI layer, while the code in `UriClientSetting` defines the same conversion in the persistence layer.  While they happen to be the same, it's just coincidence.  Therefore, I kept them independent.

One motivation for the independence is that I see the UI-related client setting stuff (including this class) moving to a different module instead of having UI and non-UI code in the same module and package.  So, at that time, it will become clearer why the (perceived) duplication is necessary.

For the same reason, I introduced `SelectionComponentFactory$ValueEncodingException`.  Even though it has the same semantics as `ClientSetting$ValueEncodingException`, the former is intended for the UI layer, whereas the latter is for the persistence layer.